postgres-protocol/src/authentication/sasl.rs

```diff
@@ -49,6 +50,11 @@ pub(crate) fn hi(str: &[u8], salt: &[u8], i: u32) -> [u8; 32] {
         for (hi, prev) in hi.iter_mut().zip(prev) {
             *hi ^= prev;
         }
+        // yield every ~250us
+        // hopefully reduces tail latencies
+        if i % 1024 == 0 {
```

Comment:
You, probably, wanted to use `_` instead.
Let's refactor this function, it's a bit confusing. argument `i` -> `iterations`.

Reply:
oh yes... good catch